.github/workflows/trigger-rulesets-test.yaml

```
@@ -0,0 +1,176 @@
+          go-version: '1.23'
+
+      - name: Install Java and Maven
+        uses: s4u/setup-maven-action@v1.18.0
+        with:
+          java-version: 17
+          java-distribution: temurin
+          maven-version: 3.9.0
+          checkout-enabled: false
+
+      - name: Checkout analyzer-lsp
+        uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)